custom border radius in BarTouchTooltipData

MattiaPispisa opened the pull request:
**IMPROVEMENT** (by @MattiaPispisa) custom border radius in `BarTouchTooltipData`, `LineTouchTooltipData`, `ScatterTouchTooltipData` #1715

imaNNeo commented on lib/src/chart/bar_chart/bar_chart_data.dart:
```diff
@@ -702,6 +705,7 @@ class BarTouchTooltipData with EquatableMixin {
   /// also you can set [fitInsideVertically] true to force it to shift inside the chart vertically.
   BarTouchTooltipData({
     double? tooltipRoundedRadius,
+    BorderRadius? tooltipBorderRadius,
```
We already have the `tooltipRoundedRadius` property. If you want to add it, please remove the old one.

imaNNeo replied:
You're right, we should keep it backward-compatible.
So can you please make the old one `tooltipRoundedRadius` deprecated? This way we can remove it in the next releases

imaNNeo commented on CHANGELOG.md:
```diff
@@ -1,3 +1,6 @@
+## UNRELEASED
+* **IMPROVEMENT** (by @MattiaPispisa) custom border radius in BarTouchTooltipData, #1715
```
You don't need to write a changelog here, please read our [contributing guideline](https://github.com/imaNNeo/fl_chart/blob/main/CONTRIBUTING.md)

MattiaPispisa replied:
I had read the guidelines, and it was recommended to start from an issue. From the ‘Mention the related issues’ section, what I did seemed correct. What should I have done differently?
Thank you.

imaNNeo replied:
Ahh, you're right. I'm sorry I forgot to update one part of the guideline.
I just updated it [here](https://github.com/imaNNeo/fl_chart/pull/1775/commits/cef2d74363c4ddda20ab22c075ead9b86fd75ad0).

As we had a lot of conflicts in the change log part, I decided to update it by myself. So you don't need to write something.

Maybe, it would be a good idea to just copy-paste yours in the PR's description. This way I can just add it.

MattiaPispisa commented:
Hi @imaNNeo, no problem.  

I removed the changelog and included what I had written in the issue description. 
I also added the annotation (deprecated) to the parameter.

imaNNeo commented:
Very nice!
Can you please add the same feature for our LineChart and ScatterChart tooltips to keep the consistency?

Also please rebase your branch.

MattiaPispisa commented:
Thank you! 

I've applied the same changes to both `LineChart` and `ScatterChart`. The implementation in `LineChart` is slightly different because the constructor is a `const`, and I wanted to keep it that way to avoid introducing a breaking change. The exposed interface remains the same.

I also made a rebased with `main` branch.

imaNNeo commented:
Everything seems good to me
Thanks for your contribution!
Hope to see more from you!
